handler/rfc8628/token_endpoint_handler.go

```diff
@@ -0,0 +1,62 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package rfc8628
+
+import (
+	"context"
+
+	"github.com/ory/fosite/handler/oauth2"
+	"github.com/ory/x/errorsx"
+
+	"github.com/ory/fosite"
+)
+
+// DeviceAuthorizeHandler is a response handler for the Device Code introduced in the Device Authorize Grant
+// as defined in https://www.rfc-editor.org/rfc/rfc8628
+type DeviceAuthorizeHandler struct {
+	DeviceStrategy DeviceCodeStrategy
+	DeviceStorage  DeviceCodeStorage
+}
+
+type DeviceCodeTokenEndpointHandler struct {
+	oauth2.GenericCodeTokenEndpointHandler
```

Comment:
Why do we need to wrap the struct here?

Reply:
This is so that we have a named struct that can be specialized instead of using the `GenericCodeTokenEndpointHandler` directly.

I did that to follow @aeneasr first review: 
https://github.com/ory/fosite/pull/701#discussion_r1059856835